core/server/worker/src/main/java/alluxio/worker/block/reviewer/Reviewer.java

```diff
@@ -0,0 +1,36 @@
+package alluxio.worker.block.reviewer;
+
+import alluxio.annotation.PublicApi;
+import alluxio.conf.PropertyKey;
+import alluxio.conf.ServerConfiguration;
+import alluxio.util.CommonUtils;
+import alluxio.worker.block.BlockMetadataView;
+import alluxio.worker.block.allocator.Allocator;
+import alluxio.worker.block.allocator.MaxFreeAllocator;
+import alluxio.worker.block.meta.StorageDirView;
+import com.google.common.base.Preconditions;
+
+@PublicApi
+public interface Reviewer {
+  boolean reviewAllocation(StorageDirView dirView);
+
+  /**
+   * Factory for {@link Reviewer}.
+   */
+  class Factory {
+
+    private Factory() {} // prevent instantiation
+
+    // TODO(jiacheng): Is it possible we need a list of reviewers in the future, each checking one criteria?
```

Comment:
@gpang Do you think it's possible that we may want a pipeline of Reviewers, instead of having only one?

Reply:
Hrmmm, I'm not sure. I don't know what a Reviewer is yet.